backend/clubs/views.py

```diff
@@ -1526,6 +1528,31 @@ def get_queryset(self):
         return Badge.objects.filter(Q(purpose="fair") | Q(label="SAC"))
 
 
+class FavoriteCalendarAPIView(APIView):
+    """
+        get: Return a .ics file of the user's favorite clubs' events.
+    """
+
+    def get(self, request, *args, **kwargs):
+        calendar = ICSCal()
+        calendar.extra.append(ICSParse.ContentLine(name="X-WR-CALNAME", value="Penn Clubs Events"))
+
+        all_events = Event.objects.filter(
+            club__favorite__person__profile__uuid_secret=kwargs["user_secretuuid"]
+        ).distinct()
+        for event in all_events:
+            e = ICSEvent()
+            e.name = event.club.name + ": " + event.name
```

Comment:
I kind of prefer the calendar name event format to use brackets for the club name, but that may be because Canvas is already in that format. I'll leave the final decision up to you!

![image](https://user-images.githubusercontent.com/4441174/96372586-7ba0b280-1135-11eb-95fe-954826795ec7.png)